packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -19,6 +19,17 @@ class NotType extends QueryPredicateOperator<NotType> {
   }
 }
 
+/**
+ * A predicate operator that returns true if the nodes range is not empty.
+ */
+class NotEmpty extends QueryPredicateOperator<NotEmpty> {
```

Comment:
That Talon Tree sitter can contain an empty matches node if there is no header. When this happens and the user has an empty cursor at the start of the document we get an empty range for the iteration scope for key/value/name/statement.